lib/src/binary_codec.dart

```diff
@@ -449,7 +453,7 @@ class PostgresBinaryDecoder<T> extends Converter<Uint8List?, T?> {
       return null;
     }
 
-    final dataType = typeMap[typeCode];
+    final dataType = type ?? typeMap[typeCode];
```

Comment:
Could this be merged with `type` in the constructor?

Reply:
We could move the map lookup into the constructor, but not without dropping `const`. I didn't want to do that since that might be breaking. If you think it's worth it, I'm happy to unify that in the constructor.

Reply:
I think it is worth it. The decoder is not part of the public API, so dropping `const` seems to be fine...